ste/sender-blockBlob.go

```diff
@@ -153,19 +153,38 @@ func (s *blockBlobSenderBase) Epilogue() {
 	// https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-storage-tiers
 	if jptm.IsLive() && s.destBlobTier != azblob.AccessTierNone {
 		// Set the latest service version from sdk as service version in the context.
-		ctxWithLatestServiceVersion := context.WithValue(jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
-		_, err := s.destBlockBlobURL.SetTier(ctxWithLatestServiceVersion, s.destBlobTier, azblob.LeaseAccessConditions{})
-		if err != nil {
-			if s.jptm.Info().S2SSrcBlobTier != azblob.AccessTierNone {
-				s.jptm.LogTransferInfo(pipeline.LogError, s.jptm.Info().Source, s.jptm.Info().Destination, "Failed to replicate blob tier at destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-				s2sAccessTierFailureLogStdout.Do(func() {
-					glcm := common.GetLifecycleMgr()
-					glcm.Error("One or more blobs have failed blob tier replication at the destination. Try transferring with the flag --s2s-preserve-access-tier=false")
-				})
+		ctxWithLatestServiceVersion := context.WithValue(s.jptm.Context(), ServiceAPIVersionOverride, azblob.ServiceVersion)
+
+		// Let's check if we can confirm we'll be able to check the destination blob's account info.
+		// A SAS token, even with write-only permissions is enough. OR, OAuth with the account owner.
+		// We can't guess that last information, so we'll take a gamble and try to get account info anyway.
+		destParts := azblob.NewBlobURLParts(s.destBlockBlobURL.URL())
+		mustGet := destParts.SAS.Encode() != ""
+
+		prepareDestAccountInfo(s.destBlockBlobURL.BlobURL, s.jptm, ctxWithLatestServiceVersion, mustGet)
+		tierAvailable := BlobTierAllowed(s.destBlobTier)
+
+		if tierAvailable {
+			_, err := s.destBlockBlobURL.SetTier(ctxWithLatestServiceVersion, s.destBlobTier, azblob.LeaseAccessConditions{})
+			if err != nil {
+				if s.jptm.Info().S2SSrcBlobTier != azblob.AccessTierNone {
+					s.jptm.LogTransferInfo(pipeline.LogError, s.jptm.Info().Source, s.jptm.Info().Destination, "Failed to replicate blob tier at destination. Try transferring with the flag --s2s-preserve-access-tier=false")
+					s2sAccessTierFailureLogStdout.Do(func() {
+						glcm := common.GetLifecycleMgr()
+						glcm.Error("One or more blobs have failed blob tier replication at the destination. Try transferring with the flag --s2s-preserve-access-tier=false")
```

Comment:
Why is this one glcm.Error but the other copy of this code has glcm.Info?  I would have thought both should be info. (If that's correct, it may indicate why avoiding the code duplication will help with maintenance).

Reply:
This really shouldn't be error anyway, since that kills azcopy.